pkg/reconciler/clusterimagepolicy/controller.go

```diff
@@ -80,18 +83,44 @@ func NewController(
 	// ConfigMap but there are no changes to the ClusterImagePolicy, it needs
 	// to be synced.
 	grCb := func(obj interface{}) {
-		logging.FromContext(ctx).Info("Doing a global resync on ClusterImagePolicies due to ConfigMap changing.")
+		logging.FromContext(ctx).Info("Doing a global resync on ClusterImagePolicies due to ConfigMap changing or resync period.")
 		impl.GlobalResync(clusterimagepolicyInformer.Informer())
 	}
-	// Resync on only ConfigMap changes that pertain to the one I care about.
+	// Resync on only ConfigMap changes that pertain to the one I care about
+	// or after a resync period.
 	// We could also fetch/construct the store and use CM watcher for it, but
 	// since we need a lister for it anyways in the reconciler, just set up
 	// the watch here.
-	configMapInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+	configMapInformer.Informer().AddEventHandlerWithResyncPeriod(cache.FilteringResourceEventHandler{
 		FilterFunc: pkgreconciler.ChainFilterFuncs(
 			pkgreconciler.NamespaceFilterFunc(system.Namespace()),
 			pkgreconciler.NameFilterFunc(config.ImagePoliciesConfigName)),
 		Handler: controller.HandleAll(grCb),
-	})
+	}, resyncPeriod(ctx))
+
 	return impl
 }
+
+// Obtain and return the policy-resync-period from config-policy-controller config map
+// Return default resync period when errors occur
+func resyncPeriod(ctx context.Context) time.Duration {
```

Comment:
I think it would be better to not pull this from the configmap (see earlier comment), but maybe make it a flag or env variable so it's clear that it will require a restart. You could even parse / stuff the parsed value into context and pull it from there. Other nice thing (I think) is that if you specify an invalid value, you can error out without resorting to logs.